lib/lightning/config/bootstrap.ex

```diff
@@ -251,16 +254,79 @@ defmodule Lightning.Config.Bootstrap do
            env!("INIT_PROJECT_FOR_NEW_USER", &Utils.ensure_boolean/1, false)
 
     # To actually send emails you need to configure the mailer to use a real
-    # adapter. You may configure the swoosh api client of your choice. We
-    # automatically configure Mailgun if an API key has been provided. See
-    # https://hexdocs.pm/swoosh/Swoosh.html#module-installation for more details.
-    if api_key = env!("MAILGUN_API_KEY", :string, nil) do
-      config :lightning, Lightning.Mailer,
-        adapter: Swoosh.Adapters.Mailgun,
-        api_key: api_key,
-        domain: env!("MAILGUN_DOMAIN", :string)
+    # adapter. You may configure the swoosh api client of your choice.
+    # See # https://hexdocs.pm/swoosh/Swoosh.html#module-installation for more details.
+    if mail_provider = env!("MAIL_PROVIDER", :string, nil) do
+      case mail_provider do
+        "local" ->
+          config :lightning, Lightning.Mailer, adapter: Swoosh.Adapters.Local
+
+        "mailgun" ->
+          config :lightning, Lightning.Mailer,
+            adapter: Swoosh.Adapters.Mailgun,
+            api_key: env!("MAILGUN_API_KEY", :string),
+            domain: env!("MAILGUN_DOMAIN", :string)
+
+        "smtp" ->
+          # TODO: HOW DO WE LET USERS PICK WHAT TO CONFIGURE HERE?
+          # https://hexdocs.pm/swoosh/Swoosh.Adapters.SMTP.html#content
+          config :lightning, Lightning.Mailer,
+            adapter: Swoosh.Adapters.SMTP,
+            username: env!("SMTP_USERNAME", :string),
+            password: env!("SMTP_PASSWORD", :string),
+            relay: env!("SMTP_RELAY", :string),
+            tls:
+              env!(
+                "SMTP_TLS",
+                fn v ->
+                  case v do
+                    "true" ->
+                      :always
+
+                    "false" ->
+                      :never
+
+                    "if_available" ->
+                      :if_available
+
+                    unknown ->
+                      raise """
+                      Unknown SMTP_TLS value: #{unknown}
+
+                      Must be one of: true, false, if_available
+                      """
+                  end
+                end,
+                :always
+              ),
+            port: env!("SMTP_PORT", :integer, 587)
+
+        unknown ->
+          raise """
+          Unknown mail provider: #{unknown}
+
+          Currently supported providers are:
+
+          - local (default)
+          - mailgun
+          - smtp
+          """
+      end
     end
 
+    # get_env(:lightning, :emails) |> IO.inspect()
+    config :lightning,
+      emails: [
+        sender_name: env!("EMAIL_SENDER_NAME", :string, "OpenFn"),
```

Comment:
@stuartc From a branding POV, would this not be better as "Lightning"? Alternatively, from a company reputation POV, if I set up a poorly configured Lightning that spams out emails, would it not be better to have the sender name as "Lightining"?

Reply:
Right now a lot of people refer to Lightning as OpenFn, so probably a better default for now.